lib/kuzzle/kuzzle.js

```diff
@@ -247,6 +253,56 @@ class Kuzzle extends KuzzleEventEmitter {
     process.exit(0);
   }
 
+  /**
+   * Execute multiple handlers only once during the lifetime of the app
+   * 
+   * @param {Array<{ id: string, handler: () => void }>} installations - Array of unique methods to execute
+   * 
+   * @returns {Promise<number>} Executed handler count
+   */
+  async install (installations) {
+    if ( !installations || !installations.length) {
+      return 0;
+    }
+    
+    let executedCount = 0;
+    const mutex = new Mutex('backend:installations');
+    await mutex.lock();
+
+    try{
+      for (const installation of installations) {
+        const alreadyInstalled = await this.ask(
+          'core:storage:private:document:exist',
+          'kuzzle',
+          'installations',
+          installation.id );
+
+        if (!alreadyInstalled) {
+          try {
+            await installation.handler();
+          }
+          catch (error) {
+            throw kerror.get( 'plugin', 'runtime', 'unexpected_installation_error', installation.id, error);
+          }    
+
+          await this.ask(
+            'core:storage:private:document:create',
+            'kuzzle',
+            'installations',
+            { handler: installation.handler.toString(), installedAt: Date.now() },
+            { id: installation.id });
+
+          executedCount += 1;
+        }
```

Comment:
As a user, I'd like to have a warning here, saying the install code has not been re-executed.

Reply:
IMHO it's not a warning because that's the expected behavior to not re-execute already executed installations

Reply:
There is already a short fedback of what have been done [here](https://github.com/kuzzleio/kuzzle/pull/2048/files#diff-ee0c987c21f421e4392b8cc04b7737eb94ff6c2b903c02c2bbfb509af36acb43R204), is that sufficient?

Reply:
I don't see why we should show something that has not been executed in the logs.

About your need @xbill82 , perhaps an API route returning all the installed code (id, timestamp) could do the trick: that way admins can see what installs have been performed, and when. When debugging their app, this should be sufficient (and more to the point) to be reassured about their install code, without having to scan lengthy log messages, and without cluttering those.

WDYT?

Reply:
That seems to me like a good idea. If @xbill82 agrees, should I open an issue on that or implement it directly in this PR?